clcache.py

```diff
@@ -1371,24 +1379,24 @@ def processCacheHit(cache, objectFile, cachekey):
         return 0, cachedArtifacts.stdout, cachedArtifacts.stderr, False
 
 
-def createManifest(manifestHash, includePaths):
-    baseDir = normalizeBaseDir(os.environ.get('CLCACHE_BASEDIR'))
```

Comment:
I found confusing that the handling of the baseDir had to be done outside of the collapseBasedirToPlaceholder or expandBasedirPlaceholder making the surrounding code more complicated. Even more passing the baseDir as argument was not used during the tests as the whole setup is done through the environment variable.

Regarding the performance, it can be extracted to a constant variable at the top of the file.